app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -25,11 +26,64 @@ def bulk_destroy
         process_response @hosts.destroy_all
       end
 
+      api :PUT, "/hosts/bulk/build", N_("Build")
+      param_group :bulk_host_ids
+      param :reboot, :bool, N_("Reboot after build. Ignored if rebuild_configuration is passed.")
+      param :rebuild_configuration, :bool, N_("Rebuild configuration only")
+      def build
+        if Foreman::Cast.to_bool(params[:rebuild_configuration])
+          rebuild_config
+        else
+          reboot = Foreman::Cast.to_bool(params[:reboot])
+          manager = BulkHostsManager.new(hosts: @hosts)
+          missed_hosts = manager.build(reboot: reboot)
+          if missed_hosts.empty?
+            process_response(true, { :message => N_("Built %{host_count} %{hosts}") % { :host_count => @hosts.count, :hosts => 'host'.pluralize(@hosts.count) }})
+          else
+            process_response(false, { :message => N_("The following hosts failed the build operation: %s") % missed_hosts.map(&:name).to_sentence})
+          end
+        end
+      end
+
+      protected
+
+      def action_permission
+        case params[:action]
+        when 'build'
+          'edit'
+        else
+          super
+        end
+      end
+
       private
 
       def find_deletable_hosts
         find_bulk_hosts(:destroy_hosts, params)
       end
+
+      def find_editable_hosts
+        find_bulk_hosts(:edit_hosts, params)
+      end
+
+      def rebuild_config
+        all_fails = BulkHostsManager.new(hosts: @hosts).rebuild_configuration
+
+        message = ''
+        all_fails.each_pair do |key, values|
+          unless values.empty?
+            message << ((N_("%{config_type} rebuild failed for host: %{host_names}.",
```

Comment:
`N_()` is for "mark to translate" while `n_()` is plural form
```suggestion
            message << ((n_("%{config_type} rebuild failed for host: %{host_names}.",
```

Reply:
@evgeni and I were talking about this the other day and looking at https://projects.theforeman.org/projects/foreman/wiki/Translating, and it was unclear which one is correct for  "I want to use the percent thingies, but not necessarily pluralize any words."